src/core/tests/widgets/test_optioncontainer.py

```diff
@@ -191,3 +191,21 @@ def test_set_window(self):
         self.op_container.window = window
         for item in self.op_container.content:
             self.assertEqual(item._content.window, window)
+
+    def test_set_tab_title(self):
+        new_label = 'New Title'
+        self.op_container.content[0].label = new_label
+        self.assertEqual(self.op_container.content[0].label, new_label)
+
+    def test_insert_tab(self):
+        self.op_container.content.insert(0, label=self.label2, widget=self.widget2)
+        self.assertEqual(self.op_container.content[0].label, self.label2)
+
+    def test_swap_tabs(self):
```

Comment:
It's not clear to me what you're actually testing for here. Is this stemming from a specific failure you've seen? What is this test picking up that a simpler "assign content to tab" test would pick up?

Reply:
The simpler "assign content to tab" test would be enough. I got carried away.

Reply:
Actually the method I was testing here (OptionList.__setitem__) is not calling any implementation method, and as such doesn't work as I expected. It just updates the toga_core data structure.

And maybe `OptionList.__setitem__` shouldn't be supported at all? Replacing tabs can be achieved with delete() and insert().

What could be interesting (and this is a different feature) is replacing the tab content. You can already change the text label of the tab, but not its contents.